dart/lib/src/protocol/sentry_exception.dart

```diff
@@ -92,4 +93,51 @@ class SentryException {
         throwable: throwable ?? this.throwable,
         unknown: unknown,
       );
+
+  @internal
+  List<SentryException>? get exceptions =>
+      _exceptions != null ? List.unmodifiable(_exceptions!) : null;
+
+  @internal
+  set exceptions(List<SentryException>? value) {
+    _exceptions = value;
+  }
+
+  @internal
+  void addException(SentryException exception) {
+    _exceptions ??= [];
+    _exceptions!.add(exception);
+  }
+
+  @internal
+  List<SentryException> flatten({int? parentId, int id = 0}) {
```

Comment:
We could consider moving this to `ExceptionGroupEventProcessor`, as it's business logic and only used there.